go/ethadapter/blockprovider.go

```diff
@@ -84,14 +89,20 @@ func (e *EthBlockProvider) streamBlocks(ctx context.Context, fromHeight *big.Int
 		case <-ctx.Done():
 			return
 		default:
-			// this will block if we're up-to-date with live blocks
-			block, err := e.fetchNextCanonicalBlock(ctx, fromHeight, latestSent)
-			if err != nil {
-				e.logger.Warn("unexpected error while preparing block to stream, will retry in 1 sec", log.ErrKey, err)
-				// todo: consider possible scenarios and what an appropriate wait time is here. Perhaps use a back-off strategy?
-				time.Sleep(10 * time.Millisecond)
-				continue
-			}
+			var block *types.Block
+			var err error
```

Comment:
I'm not quite following this "err" logic.
it's used in the lambda, also as a return

Reply:
Yeah fair, I've changed it to use a different one inside the lambda for clarity. And it was ignoring the return but now I treat it as fatal if an error manages to come out of the infinite retry.